File: exam-solutions/IST603-Exam-Answers/codes/isbn/book.py

class Book:
  """Class to represent a book with its information."""
  
  def __init__(self,title,author,publisher,city,date_of_publication, price, isbn_num):
    """
    Initializer for Book class
    
    Parameters:
      - title: Book title
      - author: Book author
      - publisher: Publisher name
      - city: City of publication
      - date_of_publication: Date of publication (string)
      - price: Book price
      - isbn_num: ISBN object
    """
    
    self.title = title
    self.author = author
    self.publisher = publisher
    self.city = city
    self.date_of_publication = date_of_publication
    self.price = price
    self.isbn_num = isbn_num
    
    
  def set_book_ISBN(self, isbn_num):
    """
    Sets the ISBN for the book.
    
    Parameters:
    - isbn_num: ISBN object
    """
    self.isbn_num = isbn_num
    
  def get_author(self):
    """
    Returns the author of the book.
    
    Returns:
    - Author name as string
    """
    return self.author
    
  def get_book_ISBN(self):
    """
    Returns the ISBN object of the book.
    
    Returns:
    - ISBN object
    """
    return self.isbn_num
  
  def print_details(self):
    """
    Prints the book information in the specified format.
    """
    print(f"Book Title: {self.title}")
    print(f"Book Author: {self.author}")
    print(f"Publisher: {self.publisher}")
    print(f"ISBN: {self.isbn_num.get_ISBN()}")

File: exam-solutions/IST603-Exam-Answers/codes/isbn/codes/isbn/Book.java
package codes.isbn;

public class Book {
  // Instance variables
  private String title;
  private String author;
  private String publisher;
  private String city;
  private String dateOfPublication;
  private double price;
  private ISBN isbnNum;

  // Constructor
  public Book(String title, String author, String publisher, String city,
      String dateOfPublication, double price, ISBN isbnNum) {
    this.title = title;
    this.author = author;
    this.publisher = publisher;
    this.city = city;
    this.dateOfPublication = dateOfPublication;
    this.price = price;
    this.isbnNum = isbnNum;
  }

  // Method to set the ISBN for the book
  public void setBookISBN(ISBN isbnNum) {
    this.isbnNum = isbnNum;
  }

  // Method to get the author of the book
  public String getAuthor() {
    return author;
  }

  // Method to get the ISBN of the book
  public ISBN getBookISBN() {
    return isbnNum;
  }

  // Method to print book details in the specified format
  public void printDetails() {
    System.out.println("Book Title: " + title);
    System.out.println("Book Author: " + author);
    System.out.println("Publisher: " + publisher);
    System.out.println("ISBN: " + isbnNum.getISBN());
    System.out.println("City: " + city);
    System.out.println("Date of Publication: " + dateOfPublication);
    System.out.println("Price: $" + price);
  }
}
